test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/UriPathParserTests.cs

```diff
@@ -198,68 +198,6 @@ public void ParsePathIgnoresQueryStringEvenWhenItHasSlashes()
         }
 
-        [Theory]
-        [InlineData("root", new[] { "root" })]
-        [InlineData("root:/", new[] { "root", ":" })]
-        [InlineData("root:/:", new[] { "root", "::" })]
-        [InlineData("root:/abc", new[] { "root", ":abc" })]
-        [InlineData("root:/abc:", new[] { "root", ":abc:" })]
-        [InlineData("root:/::/", new[] { "root", "::", ":" })]
-        [InlineData("root:/:/:/", new[] { "root", "::", ":" })]
-        [InlineData("root:/:///////////:/", new[] { "root", "::", ":" })]
-        [InlineData("root:/::/:", new[] { "root", "::", "::" })]
-        [InlineData("root:/abc:/property", new[] { "root", ":abc:", "property" })]
-        [InlineData("root:/abc:/property:/", new[] { "root", ":abc:", "property", ":" })]
-        [InlineData("root:/abc:/property:/:", new[] { "root", ":abc:", "property", "::" })]
-        [InlineData("root:/abc:/property:/::/", new[] { "root", ":abc:", "property", "::", ":" })]
-        [InlineData("root:/:/property", new[] { "root", "::", "property" })]
-        [InlineData("root:/photos/2018/February", new[] { "root", ":photos/2018/February" } )]
-        [InlineData("root:/photos%2f2018%2fFebruary", new[] { "root", ":photos%2f2018%2fFebruary" } )]
-        [InlineData("root:/photos%2f2018%2fFebruary/Others", new[] { "root", ":photos%2f2018%2fFebruary/Others" } )]
-        [InlineData("root:/photos%2f2018%2f/////February/Others", new[] { "root", ":photos%2f2018%2f/////February/Others" } )]
-        [InlineData("root:/photos/2018%2fFebruary:/permissions", new[] { "root", ":photos/2018%2fFebruary:", "permissions" } )]
-        [InlineData("root:/photos:2018:/permissions:/abc", new[] { "root", ":photos:2018:", "permissions", ":abc" } )]
-        [InlineData("root:/photos::::::2018:/permissions:/abc", new[] { "root", ":photos::::::2018:", "permissions", ":abc" })]
-        [InlineData("root:/photos/::::::2018:/permissions:/abc", new[] { "root", ":photos/::::::2018:", "permissions", ":abc" })]
-        [InlineData("EntitySet('key'):/xyz:/perm", new[] { "EntitySet('key')", ":xyz:", "perm" } )]
-        [InlineData("EntitySet('key'):/xyz::/perm", new[] { "EntitySet('key')", ":xyz:", ":perm" } )]
-        [InlineData("EntitySet('key'):/xyz/abc:", new[] { "EntitySet('key')", ":xyz/abc:" } )]
-        [InlineData("EntitySet('key%2fvalue'):/xyz", new[] { "EntitySet('key%2fvalue')", ":xyz" })]
-        [InlineData("EntitySet/key%2fvalue:/xyz", new[] { "EntitySet", "key%2fvalue", ":xyz" })]
-        [InlineData("EntitySet/key:%2fvalue:/xyz", new[] { "EntitySet", "key:%2fvalue", ":xyz" })]
-        [InlineData("EntitySet('key'):/xyz::/perm", new[] { "EntitySet('key')", ":xyz:", ":perm" })]
-        [InlineData("EntitySet('key'):/xyz:/:/perm", new[] { "EntitySet('key')", ":xyz:", ":perm" })]
-        [InlineData(":/xyz::/perm", new[] { ":xyz:", ":perm" })]
-        [InlineData(":/xyz:/:/perm", new[] { ":xyz:", ":perm" })]
-        [InlineData(":/xyz://///:/perm", new[] { ":xyz:", ":perm" })]

```

Comment:
Why are all variations of this test removed? Aren't most of these still valid? Any that are now invalid should have added negative tests.

Reply:
+1, we can change the expected to make the test valid again. 
Because all the patterns are valid case, we can use this test cases to see the ParsePath works expected for these patterns. 


---
In reply to: [366144313](https://github.com/OData/odata.net/pull/1621#discussion_r366144313) [](ancestors = 366144313)